File: problems_001to025/problem_022/solution.py
import os
import sys

PROJECT_PATH = os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
sys.path.append(PROJECT_PATH)

from measurement import Stopwatch

PROBLEM_PATH = os.path.dirname(os.path.abspath(__file__))


class Solution(Stopwatch):
    def __init__(self):
        super().__init__()

    def execute(self):
        answer = 0
        stream = open(f'{PROBLEM_PATH}/names.txt', 'r')
        names = stream.readline().strip().split(',')
        names.sort()
        for order, name in enumerate(names, start=1):
            name = name.strip('"')
            worth = sum(ord(letter)-ord('A')+1 for letter in name)
            answer += order * worth
        return answer

if __name__ == '__main__':
    soln = Solution()
    result = soln.execute()
    print(result)

File: problems_001to025/problem_022/solution.ts
import fs from 'fs';
import path from 'path';

import { SolutionFunction } from '@/lib/types';
import { sum } from '@/lib/toolbox';

const solution: SolutionFunction = () => {
  let answer: ReturnType<SolutionFunction> = 0;

  const source = fs.readFileSync(path.resolve(__dirname, 'names.txt'));
  const names: string[] = JSON.parse(`[${source.toString().trim()}]`);

  names.sort();

  answer = sum(names.map((name, index) => {
    const order = index + 1;
    const worth = sum(name.split('').map((letter) => (letter.charCodeAt(0) - 'A'.charCodeAt(0) + 1)));

    return order * worth;
  }));

  return answer;
};

export default solution;
